estate/models/estate_property.py

```diff
@@ -110,9 +110,10 @@ def action_sold_property(self):
         for record in self:
             if record.status == "canceled":
                 raise UserError("Canceled properties cannot be sold.")
-            if record.status == "new":
+            offers = record.offer_ids.filtered(lambda offer: offer.status == "accepted")
+            if record.status == "new" and not offers:
```

Comment:
There may be offers that are not yet accepted, so I don't see it as right that you only validate the accepted ones at this point. You have to check if there aren't any offers.

```suggestion
            if record.status == "new" and not record.offer_ids:
```